pkg/operator/ceph/cluster/watcher.go

```diff
@@ -604,8 +449,8 @@ func (c *clientCluster) unfenceAndDeleteNetworkFence(ctx context.Context, node c
 			return false, err
 		}
 
-		if networkFence.Spec.FenceState != addonsv1alpha1.Unfenced {
-			logger.Infof("waiting for network fence CR %s to get in %s state before deletion", networkFence.Name, addonsv1alpha1.Unfenced)
+		if networkFence.Status.Message != addonsv1alpha1.UnFenceOperationSuccessfulMessage {
```

Comment:
```suggestion
		if networkFence.Status.Message != addonsv1alpha1.UnFenceOperationSuccessfulMessage {
```

This check requires csi-addons v0.8.0